### PR TITLE
examl +

### PR DESCRIPTION
**Pull request by manifestoso**
examl new pkg

### CODE REVIEW

File: var/spack/repos/builtin/packages/examl/package.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    def build(self, spec, prefix):
+        #####################
+        # Build Directories #
+        #####################
+        cd('examl')
+        make('-f', 'Makefile.SSE3.gcc')
+        cd('../parser')
+        make('-f', 'Makefile.SSE3.gcc')
+        cd('..')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/examl/package.py]:
```suggestion
        with working_dir('examl'):
            make('-f', 'Makefile.SSE3.gcc')
        with working_dir('parser'):
            make('-f', 'Makefile.SSE3.gcc')
```

**Review by adamjstewart — COMMENTED** [review]
Does this only work for the GCC compiler?

File: var/spack/repos/builtin/packages/examl/package.py (hunk 1 above)
Comment (manifestoso) [reply var/spack/repos/builtin/packages/examl/package.py]: Thx @adamjstewart 

Also, do we have a doc for all the defined system variables like `working_dir` etc?

**Comment by manifestoso** [comment]
> 
> 
> Does this only work for the GCC compiler?

Hmm, `icc` supports only partical at the moment.

File: var/spack/repos/builtin/packages/examl/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/examl/package.py]: Not really. We have the [Packaging Guide](https://spack.readthedocs.io/en/latest/packaging_guide.html) which has many things. If you're looking for a list of all things present in a `package.py` file, I would look in `lib/spack/spack/pkgkit.py` or `lib/spack/spack/build_environment.py`. Many of these things like `working_dir` are defined in `lib/spack/llnl/util/filesystem.py`.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**